opengovsg/postmangovsg
# chore: fix build warnings

## Problem

There are a few build warnings that come up in Amplify. These are all minor warnings which require zero functional changes.

## Solution

**Improvements**:

- Move the definition of custom headers for Amplify from `amplify.yml` to `customHttp.yml` (ref: https://docs.aws.amazon.com/amplify/latest/userguide/custom-headers.html#migrate-custom-headers)
- Explicitly specify --no-rewrite for source map uploads to Sentry

## Before & After Screenshots

There are no functional or visual changes.

## Tests

These changes have been verified on the branch deployed at https://fix-build-warnings.d3k3b7zs07zjio.amplifyapp.com/

**Test 1: Amplify custom headers should remain unchanged**
1. Deploy the change to Amplify.
2. Ensure that the following warnings no longer appear:
```
2021-03-30T01:45:46.599Z [WARNING]: [DEPRECATION WARNING] using amplify.yml to modify custom headers is being deprecated.
2021-03-30T01:45:46.599Z [WARNING]: Going forward, in order to configure your custom headers via deploy, please use customHttp.yml instead.
2021-03-30T01:45:46.599Z [WARNING]: Learn more: https://docs.aws.amazon.com/amplify/latest/userguide/custom-headers.html
```
3. Visit the landing page. (this should be the same for any page)
4. Under the network tab of Chrome devtools, look at the response headers of the initial GET request.
5. Ensure that the response headers that are specified in `customHttp.yml` are exactly the same as those in the GET request.
6. Also check the response headers for a static file like https://fix-build-warnings.d3k3b7zs07zjio.amplifyapp.com/static/js/runtime-main.f46cbea8.js

**Test 2: Sentry warning no longer appears**
1. Deploy the change on Amplify.
2. Verify that the following build warnings no longer appear:
```
2021-03-30T01:45:40.429Z [WARNING]: WARN    2021-03-30 01:45:40.429564075 +00:00 The default --no-rewrite will disappear. Please specify --rewrite or --no-rewrite explicitly during sourcemap upload.
```
